depthimage_to_navscan_rgbd/src/depthimage_to_navscan_rgbd.cpp

```diff
@@ -24,23 +24,61 @@ int main(int argc, char** argv)
     }
 
     DepthSensorIntegrator depthSensorIntegrator;
-    //TODO configure DepthSensorIntegrator
+    // get depth sensor integrator parameters
+    int num_samples;
+    double slope_threshold;
+    double min_distance;
+    double max_distance;
+    int slope_window_size;
+
+    if (!nh.getParam("slope_threshold", slope_threshold)) {
+        ROS_FATAL("[Depthimage to Navscan RGBD] could not read slope_threshold from parameter server");
+        return 1;
+    }
+    if (!nh.getParam("min_distance", min_distance)) {
+        ROS_FATAL("[Depthimage to Navscan RGBD] could not read min_distance from parameter server");
+        return 1;
+    }
+    if (!nh.getParam("max_distance", max_distance)) {
+        ROS_FATAL("[Depthimage to Navscan RGBD] could not read max_distance from parameter server");
+        return 1;
+    }
+    if (!nh.getParam("slope_window_size", slope_window_size)) {
+        ROS_FATAL("[Depthimage to Navscan RGBD] could not read slope_window_size from parameter server");
+        return 1;
+    }
+    if (!nh.getParam("num_samples", num_samples)) {
+        ROS_FATAL("[Depthimage to Navscan RGBD] could not read num_samples from parameter server");
+        return 1;
+    }
+
+    depthSensorIntegrator.initialize(slope_threshold, min_distance, max_distance, num_samples, slope_window_size);
+    ROS_INFO("[Depthimage to Navscan RGBD] initialised depth sensor integrator with:\n"
+              " slope_threshold: %f, min_distance: %f, max_distance: %f, num_samples: %i, slope_window_size: %i",
+              slope_threshold, min_distance, max_distance, num_samples, slope_window_size);
 
     ImageBuffer image_buffer;
     image_buffer.initialize(rgbd_topic, map_frame);
 
-    ros::Publisher pointcloud2_publisher = nh.advertise<sensor_msgs::PointCloud2>("navigation/cloud", 20);
+    ros::Publisher pointcloud2_publisher = nh.advertise<sensor_msgs::PointCloud2>("navscan", 20);
 
     while (ros::ok())
     {
-        if (!depthSensorIntegrator.isInitialized())
-            continue;
-
         rgbd::ImageConstPtr image;
         geo::Pose3D sensor_pose;
         if (!image_buffer.nextImage(image, sensor_pose))
             continue;
 
+        if (!depthSensorIntegrator.isInitialized()) {
+            ROS_INFO("[Depthimage to Navscan RGBD] configuring camera model");
```

Comment:
Reconsider the log level, I think debug is better.

Reply:
I think info is fine, especially since it is only written once when the first image is received

Reply:
Oh when it logs to the logfile, info's don't show. So than it is fine, but if we would log all configure hooks of our software, we have a very long log 😉